services/app-web/src/components/ExpandableText/ExpandableText.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useLayoutEffect, useState } from 'react'
+import { Button } from '@trussworks/react-uswds'
+import styles from './ExpandableText.module.scss'
+
+export type ExpandableTextProps = {
+    clamp: React.ReactElement | string // you can pass in an element with nested <span> or simple string
+    clampedLines?: number
+}
+
+export const ExpandableText = ({
+    clamp,
+    clampedLines = 2,
+    ...props
+}: ExpandableTextProps &
+    React.HTMLAttributes<HTMLDivElement>): React.ReactElement => {
+    //We are using React.useRef because jest spyOn useRef only works like this.
+    const textRef = React.useRef<HTMLDivElement>(null)
+    const [showMore, setShowMore] = useState<boolean>(false)
+    const [showMoreButton, setShowMoreButton] = useState<boolean>(false)
+
+    useLayoutEffect(() => {
+        const ref = textRef?.current
+        if (
+            ref &&
+            (ref.offsetHeight < ref.scrollHeight ||
+                ref.offsetWidth < ref.scrollWidth)
+        ) {
+            if (clampedLines) {
+                ref.style.webkitLineClamp = clampedLines.toString()
```

Comment:
neat! I didn't know this was a browser built-in

Reply:
For chrome/safari its been there for awhile, firefox added support a lot later and it seems like its not fully there it seems Jason saw weirdness with this even in Storybook.  Eventually CSS will have `line-clamp` officially (without need for browser specific polyfill) but its out of [draft](https://www.w3.org/TR/css-overflow-3/#line-clamp) yet.